srcts/src/components/sidebar.ts

```diff
@@ -28,20 +63,75 @@ class Sidebar {
     TRANSITIONING: "transitioning",
   };
 
-  public static initCollapsible(container: HTMLElement): void {
-    // Signal that this layout is initialized by removing the init attribute
-    container.removeAttribute("data-bslib-sidebar-init");
+  private static onReadyScheduled = false;
+  private static instanceMap: WeakMap<HTMLElement, Sidebar> = new WeakMap();
+
+  public static getInstance(el: HTMLElement): Sidebar | undefined {
+    return Sidebar.instanceMap.get(el);
+  }
+
+  public static initCollapsibleAll(scope: Document | HTMLElement): void {
+    if (document.readyState === "loading") {
+      if (!Sidebar.onReadyScheduled) {
+        Sidebar.onReadyScheduled = true;
+        document.addEventListener("DOMContentLoaded", () =>
+          Sidebar.initCollapsibleAll(document)
+        );
+      }
+      return;
+    }
 
-    Sidebar._initSidebarCounters(container);
-    Sidebar._initDesktop(container);
+    const initSelector = "[data-bslib-sidebar-init]";
+    let scopeMatches = false;
+    if (scope instanceof HTMLElement) {
+      scopeMatches = scope.matches(initSelector);
+    }
+    if (!(scopeMatches || scope.querySelector(initSelector))) {
+      // no sidebars to initialize
+      return;
+    }
+
+    const containers = Array.from(scope.querySelectorAll(initSelector));
+    if (scopeMatches && scope instanceof HTMLElement) {
+      containers.unshift(scope);
+    }
+
+    containers.forEach((container) => new Sidebar(container as HTMLElement));
+  }
+
+  private _initEventListeners(): void {
+    const { sidebar, toggle } = this.layout;
+
+    toggle.addEventListener("click", (ev) => {
+      ev.preventDefault();
+      this.toggle("toggle");
+    });
+
+    // Once the collapse transition completes (on the collapse toggle icon, which is
+    // always guaranteed to transition), then remove the transitioning class
+    toggle
+      .querySelector(".collapse-icon")
+      ?.addEventListener("transitionend", () => {
+        setTimeout(
+          () => {
+            this._finalizeState();
+            $(sidebar).trigger("toggleCollapse.sidebarInputBinding");
+          },
+          // Add a small delay before finalizing the closed state, otherwise
+          // this happens just before the sidebar reaches the final state and
+          // the sidebar disappears abruptly.
+          this.isClosed ? 100 : 0
+        );
+      });
```

Comment:
Try to get rid of the `setTimeout` by matching up the transition times

Reply:
These transitions are now aligned in 7143671a67ae241163b305b0d8433667eb6ecc41. I played with the collapse icon easing function a little and we now speed it up considerably on mobile. Overall the effect is nice.